graphql_compiler/interpreter/typedefs.py

```diff
@@ -156,8 +154,72 @@ def get_tokens_of_type(
         neighbor_hints: Optional[Collection[Tuple[EdgeInfo, NeighborHint]]] = None,
         **hints: Any,
     ) -> Iterable[DataToken]:
-        """Produce an iterable of tokens for the specified type name."""
-        # TODO(predrag): Add more docs in an upcoming PR.
+        """Produce an iterable of tokens for the specified type name.
+
+        This function is used by the interpreter library to get the initial data with which
+        the process of query execution begins.
+
+        Consider the following example schema:
+        ***
+        schema {
+            query: RootSchemaQuery
+        }
+
+        < ... some default GraphQL compiler directives and scalar type definitions here ... >
+
+        type Foo {
+            < ... some fields here ... >
+        }
+
+        < ... perhaps other type definitions here ... >
+
+        type RootSchemaQuery {
+            # This is the root query type for the schema, as defined at the top of the schema.
+            Foo: [Foo]
+        }
+        ***
+
+        Per the GraphQL specification, since the definition of RootSchemaQuery only contains the
+        type named Foo, queries must start by querying for Foo in order to be valid for the schema:
+        {
+            Foo {
+                < stuff here >
+            }
+        }
+
+        To compute the results for such a query, the interpreter would call get_tokens_of_type()
+        with "Foo" as the type_name value. As get_tokens_of_type() yields tokens, the interpreter
+        uses those tokens to perform the rest of the query via the remaining API methods.
```

Comment:
```suggestion
        uses those tokens to perform the rest of the query via the remaining interpreter API methods.
```
If the reader's thinking about how to implement an adapter over some other API, the clarification may help here.